apps/astarte_appengine_api/lib/astarte_appengine_api/device/queries.ex

```diff
@@ -234,20 +240,33 @@ defmodule Astarte.AppEngine.API.Device.Queries do
       # TODO: should we handle this situation?
     end
 
-    # TODO: :reception_timestamp_submillis is just a place holder right now
-    unset_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(
-        "DELETE FROM #{interface_descriptor.storage} WHERE device_id=:device_id AND interface_id=:interface_id AND endpoint_id=:endpoint_id AND path=:path"
-      )
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, endpoint.endpoint_id)
-      |> DatabaseQuery.put(:path, path)
+    %InterfaceDescriptor{interface_id: interface_id, storage: storage} = interface_descriptor
+    keyspace_name = Realm.keyspace_name(realm_name)
 
-    DatabaseQuery.call!(db_client, unset_query)
+    q =
+      from v in storage,
+        prefix: ^keyspace_name,
+        where:
+          v.device_id == ^device_id and v.interface_id == ^interface_id and
+            v.endpoint_id == ^endpoint and v.path == ^path
 
-    :ok
+    case Repo.delete_all(q) do
+      {n, _} when is_number(n) ->
+        {:ok}
+
+      nil ->
+        {:error, :not_found}
+
+      {:error, reason} ->
+        _ =
+          Logger.warning(
+            "Could not unset value for  #{Device.encode_device_id(device_id)} in #{storage}}",
+            realm: "realm",
+            tag: "cant_unset"
+          )
+
+        {:error, reason}
+    end
```

Comment:
`Repo.delete_all` always returns `{count, records}`, so the latter two cases will never match and can be removed.
On the first case instead, the previous function returned `:ok`, not `{:ok}`.